scripts/check_compliance.py

```diff
@@ -571,7 +573,7 @@ class GitLint(ComplianceTest):
     _doc = "https://docs.zephyrproject.org/latest/contribute/#commit-guidelines"
 
     def run(self):
-        self.prepare()
+        self.prepare(os.path.join(os.getcwd(), '[.gitlint]'))
```

Comment:
What's with the brackets?

Reply:
It means "optional" like in man pages. Any better idea?

Reply:
Now I'm even more confused... so the output says this:

```
Running Gitlint          in /home/joe/zephyr-ci-tools/scripts/[.gitlint]..
```

Does that mean it looks for both `/home/joe/zephyr-ci-tools/scripts/[.gitlint]` and `/home/joe/zephyr-ci-tools/scripts`?

Reply:
The GitLint check looks for just that `.gitlint` file and falls back on defaults if not found. So if you run it from a subdirectory like I did then it runs "successfully" except with the default settings instead of the repo settings. See example above. This should a one-line fix but later; I really don't want to bury/hide any fix now in this relatively large but zero functional change. So the main purpose is to draw the user's attention/suspicion on the hidden `.gitlint` file and it seems it succeeded in at least that :-) Before I wrote this patch I had completely forgotten about this hidden file myself! Then I didn't expect another .gitlint to be found in zephyr-ci-tools/ in addition to the one in ZEPHYR_BASE/.

So yes, I agree it's struggling to cram a bit too much information in a very limited space. On the other hand the entire `class GitLint` is 20 lines long including comments if anyone wants to get to the bottom of this.

Reply:
🤷‍♂  i find this confusing but I don't think it should block the merge.

Reply:
Thanks, I promise to submit the one-line change that makes gitlint more robust once this is merged (and proved not to cause any problem for a few days)

Reply:
Could spell it out maybe, like

```
Running Gitlint          in /home/joe/zephyr-ci-tools/scripts (possibly with settings from a .gitlint file in that directory)
```

Or the code could check if there's actually a `.gitlint` in there and alter the message.

Reply:
As promised, gitlint fix submitted at PR #60